enhancements/sig-architecture/README.md

```diff
@@ -0,0 +1,430 @@
+```shell
+arn:aws:iam::<account-id>:role/managed_<managed-cluster-name>_<hub-account-id>_<hub-cluster-name>
+```
+
+#### Graceful resource cleanup on both hub and managed cluster when managed cluster triggers unjoin
+
+The `ManagedCluster` will have a new boolean field `clientUnjoinsHub` with the default value of false. The `clusteradm unjoin` command by managed cluster admin will update the `ManagedCluster` on hub and set the `spec.clientUnjoinsHub` to true, which will trigger the new `aws-irsa` controller on hub to clean-up all the managed cluster specific resources, like IAM roles and policies on hub, and ultimately delete the `ManagedCluster`.
+
+or
+
+The unjoin command will update the `ManagedCluster` on hub and add a new condition `ManagedClusterUnjoined`, which will trigger the new `aws-irsa` controller on hub to clean-up all the managed cluster specific resources, like IAM roles and policies on hub, and ultimately delete the `ManagedCluster`.
+
+
+#### Managed cluster permission isolation on hub
+
+Every managed cluster will have a separate role on both managed cluster as well as hub IAM. Every managed cluster role on hub will allow access to managed cluster's namespace only. All roles will follow a standard naming convention, and the IAM trust policies allow roles to be assumed by relevant resources (pods, clusters, etc) only.
+
+This will ensure that roles can only be assumed by the consumers they are meant for, such that one managed cluster cannot assume roles and access resources belonging to other managed cluster
+
+### Workflow Details
+
+This is to describe the various process of cluster join
+
+Actors:
+1. cluster-admin on managed cluster
+2. AWS IAM admin on managed cluster AWS account
+3. cluster-admin on hub cluster
+4. AWS IAM admin on hub cluster AWS account
+5. hub controller
+6. agent on managed cluster
+
+Some rules on cluster join:
+- The name of the cluster must be globally unique on hub and conforms to dns label format.
+
+#### Managed Cluster Prerequisites
+
+An IAM role named using following template:
+```
+arn:aws:iam::<account-id>:role/managed_<managed-cluster-name>_<hub-account-id>_<hub-cluster-name>
```

Comment:
Since we are able to find out that the principal tags work, We can actually shorten the name of the role.